### PR TITLE
feat(selinux): Get rid of all file WX of Trivalent's Selinux trivalent_domain

### PR DESCRIPTION
**Pull request by PhysicsIsAwesome**
This gets rid of all file WX of Trivalent's Selinux trivalent_domain I could find:

- cache_home_t
- chrome_sandbox_home_t
- config_home_t
- trivalent_home_t
- trivalent_domain itself

It introduces a new on-by-default tunable trivalent_exec_flatpaks so people can disable execution of Flatpaks. Execution is needed for certain extensions being able to talk to their app counterpart like Keepassxc.

Additionally there is a tunable trivalent_drm which is on by-default. This allows file WX and needs to be turned off by users who want W^X at the expense of not being able to use DRM restricted content.

Testing done:

- Started Trivalent
- Visited a few websites including video content (Youtube)
- Checked AVC logs
- Deleted trivalent's folder in .config
- Might need additional testing for edge cases
- RPM spec file is untested, since it would take a very long time to build Trivalent on my device and I have little experience with spec files. Review and testing is needed.

Basically https://github.com/secureblue/secureblue/pull/2029 plus the tmpfiles.d file

### CODE REVIEW

**Review by RoyalOughtness — DISMISSED** [review]

File: build/trivalent-drm-fix-secontexts.conf [hunk 1]
```diff
@@ -0,0 +1 @@
+Z %h/.config/trivalent/WidevineCdm
```
Comment (HastD) [inline build/trivalent-drm-fix-secontexts.conf:1]: I think this should be handled by the distro (that is, secureblue could add this tmpfiles.d configuration), not by the RPM package.

Comment (RKNF404) [reply build/trivalent-drm-fix-secontexts.conf:1]: Wouldnt that cause issues for non-secureblue users?

Comment (HastD) [reply build/trivalent-drm-fix-secontexts.conf:1]: Yeah, on second thought I think this is okay.

File: build/trivalent.spec [hunk 2]
```diff
@@ -640,6 +641,8 @@ cp -a %{SOURCE9} %{buildroot}%{_datadir}/gnome-control-center/default-apps/
 %if 0%{?with_selinux}
 install -Dp -m 0644 -t %{buildroot}%{_datadir}/selinux/packages/%{selinuxtype} %{modulename}.pp.bz2
 install -Dp -m 0644 -t %{buildroot}%{_datadir}/selinux/devel/include/distributed selinux/%{modulename}.if
+mkdir -p %{buildroot}%{_datadir}/user-tmpfiles.d
+install -Dp -m 0644 %{SOURCE24} %{buildroot}%{_datadir}/user-tmpfiles.d/%{modulename}-drm-fix-secontexts.conf
```
Comment (HastD) [inline build/trivalent.spec]: The `mkdir` is redundant here; `install -D` already creates the target directory if needed.

**Comment by RoyalOughtness** [comment]
@PhysicsIsAwesome did you confirm that protected content still works with this?

File: build/trivalent.spec [hunk 3]
```diff
@@ -719,6 +724,6 @@ fi
 %{_datadir}/selinux/packages/%{selinuxtype}/%{modulename}.pp.*
 %{_datadir}/selinux/devel/include/distributed/%{modulename}.if
 %ghost %verify(not md5 size mode mtime) %{_selinux_store_path}/%{selinuxtype}/active/modules/200/%{modulename}
-
+%{_datadir}/user-tmpfiles.d/%{modulename}-drm-fix-secontexts.conf
```
Comment (HastD) [inline build/trivalent.spec]: Instead of `%{_datadir}/user-tmpfiles.d`, add `BuildRequires: systemd-rpm-macros` and use `%{_user_tmpfilesdir}` to refer to the directory.

File: build/trivalent.spec (hunk 2 above)
Comment (HastD) [inline build/trivalent.spec]: Also change `%{_datadir}/user-tmpfiles.d` to `%{_user_tmpfilesdir}` here.

**Review by HastD — CHANGES_REQUESTED** [review]
The policy changes look fine to me; just a few nitpicks about the packaging.

File: build/trivalent.spec [hunk 4]
```diff
@@ -86,6 +86,7 @@ Source18: %{chromium_name}256.png
 Source21: %{chromium_name}.fc
 Source22: %{chromium_name}.if
 Source23: %{chromium_name}.te
+Source24: %{chromium_name}-drm-fix-secontexts.conf
```
Comment (RKNF404) [inline build/trivalent.spec:90]: Shouldn't this be gated behind `enable_proprietary_codecs`?
As with everything else related to widevine with this PR in the spec.

File: build/trivalent-drm-fix-secontexts.conf (hunk 1 above)
Comment (PhysicsIsAwesome) [reply build/trivalent-drm-fix-secontexts.conf:1]: I would also prefer it to be in Trivalent's RPM package

File: build/trivalent.spec (hunk 2 above)
Comment (PhysicsIsAwesome) [reply build/trivalent.spec]: Thx. I removed the `mkdir`

File: build/trivalent.spec (hunk 3 above)
Comment (PhysicsIsAwesome) [reply build/trivalent.spec]: Thx. Done

File: build/trivalent.spec (hunk 4 above)
Comment (PhysicsIsAwesome) [reply build/trivalent.spec:90]: I put it behind `enable_proprietary_codecs`. Would you mind checking again, if it is ok?

**Comment by PhysicsIsAwesome** [comment]
> @PhysicsIsAwesome did you confirm that protected content still works with this?

Yes. With `trivalent_drm` set to `on`, it works as expected. With `off`, websites still sometimes report that drm works, but, if you actually try to play drm protected content, it does not work.

File: build/trivalent.spec (hunk 4 above)
Comment (RKNF404) [reply build/trivalent.spec:90]: Lgtm

**Review by HastD — APPROVED** [review]

**Review by RKNF404 — APPROVED** [review]
Lgtm

**Review by WavyEbuilder — APPROVED** [review]

**Review by RoyalOughtness — APPROVED** [review]
